frontend/talentsearch/src/js/components/myStatusForm/MyStatusForm.tsx

```diff
@@ -0,0 +1,283 @@
+  initialData: GetMystatusQuery | undefined;
+  handleMyStatus: (
+    id: string,
+    data: UpdateUserAsUserInput,
+  ) => Promise<UpdateMystatusMutation["updateUserAsUser"]>;
+}
+
+export const MyStatusForm: React.FC<MyStatusFormProps> = ({
+  initialData,
+  handleMyStatus,
+}) => {
+  const intl = useIntl();
+
+  function bold(msg: string) {
+    return <span data-h2-font-weight="b(700)">{msg}</span>;
+  }
+
+  const firstName = initialData?.me?.firstName;
+  const lastName = initialData?.me?.lastName;
+  const email = initialData?.me?.email;
+  const telephone = initialData?.me?.telephone;
+  const preferredLang = initialData?.me?.preferredLang;
+  const currentProvince = initialData?.me?.currentProvince;
+  const currentCity = initialData?.me?.currentCity;
+  const lookingForEnglish = initialData?.me?.lookingForEnglish;
+  const lookingForFrench = initialData?.me?.lookingForEnglish;
+  const lookingForBilingual = initialData?.me?.lookingForBilingual;
+  const isGovEmployee = initialData?.me?.isGovEmployee;
+  const locationPreferences = initialData?.me?.locationPreferences;
+  const wouldAcceptTemporary = initialData?.me?.wouldAcceptTemporary;
+  const expectedSalary = initialData?.me?.expectedSalary;
+
+  const JobLookingStatusSortOrder = [
+    JobLookingStatus.ActivelyLooking,
+    JobLookingStatus.OpenToOpportunities,
+    JobLookingStatus.Inactive,
+  ];
+  const dataToFormValues = (
+    data?: GetMystatusQuery | undefined,
+  ): FormValues => {
+    return {
+      jobLookingStatus: data?.me?.jobLookingStatus,
+    };
+  };
+  const formValuesToSubmitData = (
+    values: FormValues,
+  ): UpdateUserAsUserInput => {
+    return {
+      jobLookingStatus: values.jobLookingStatus,
+    };
+  };
+
+  const methods = useForm<FormValues>({
+    defaultValues: dataToFormValues(initialData),
+  });
+  const [prevFormValues, setPrevFormValues] = React.useState(
+    dataToFormValues(initialData),
+  );
+  const { control } = methods;
+  const formValues = useWatch({ control, name: "jobLookingStatus" });
+
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    if (initialData?.me) {
+      await handleMyStatus(initialData.me?.id, formValuesToSubmitData(data));
+    }
+  };
+
+  // Whenever form values change (with some debounce allowance), call updateCandidateFilter
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const submitDebounced = useCallback(
+    debounce(() => {
+      if (!(formValues === prevFormValues.jobLookingStatus))
+        onSubmit({ jobLookingStatus: formValues });
+      setPrevFormValues({ jobLookingStatus: formValues });
+    }, 200),
+    [formValues],
+  );
+
+  // Use deep comparison to prevent infinite re-rendering
+  useDeepCompareEffect(() => {
+    submitDebounced();
+    return () => {
+      // Clear debounce timer when component unmounts
+      submitDebounced.clear();
+    };
+  }, [formValues, submitDebounced]);
+
+  let isFormActive = true;
```

Comment:
In the issue description it says that "overall complete check" should happen on the backend.  Time to get your hands dirty with some PHP!  :grin:

Reply:
Ha Ha! It make sense. I could see some graphql error messages those are not user friedly.

Reply:
@petertgiles I couldn't see that in the issue though.

Reply:
Done! May need some changes!